src/sentry/api/endpoints/organization_metrics.py

```diff
@@ -304,3 +306,107 @@ def get_result(self, limit, cursor=None):
+            RateLimitCategory.USER: default_rate_limit,
+            RateLimitCategory.ORGANIZATION: default_rate_limit,
+        },
+    }
+
+    # Number of groups returned by default for each query.
+    default_limit = 20
+
+    def _validate_order(self, order: str | None) -> FormulaOrder | None:
+        if order is None:
+            return None
+
+        formula_order = FormulaOrder.from_string(order)
+        if formula_order is None:
+            order_choices = [v.value for v in FormulaOrder]
+            raise InvalidMetricsQueryError(
+                f"The provided `order` is not a valid, only {order_choices} are supported"
+            )
+
+        return formula_order
+
+    def _validate_limit(self, limit: str | None) -> int:
+        if not limit:
+            return self.default_limit
+
+        try:
+            return int(limit)
+        except ValueError:
+            raise InvalidMetricsQueryError(
+                "The provided `limit` is not valid, an integer is required"
+            )
+
+    def _interval_from_request(self, request: Request) -> int:
+        """
+        Extracts the interval of the query from the request payload.
+        """
+        interval = parse_stats_period(request.data.get("interval", "1h"))
+        return int(3600 if interval is None else interval.total_seconds())
+
+    def _metrics_queries_plan_from_request(self, request: Request) -> MetricsQueriesPlan:
+        """
+        Extracts the metrics queries plan from the request payload.
+        """
+        metrics_queries_plan = MetricsQueriesPlan()
+
+        queries = request.data.get("queries") or []
+        for query in queries:
+            metrics_queries_plan.declare_query(name=query["name"], mql=query["mql"])
+
+        formulas = request.data.get("formulas") or []
+        for formula in formulas:
+            metrics_queries_plan.apply_formula(
+                mql=formula["mql"],
+                order=self._validate_order(formula.get("order")),
+                limit=self._validate_limit(formula.get("limit")),
+            )
+
+        return metrics_queries_plan
+
+    def post(self, request: Request, organization) -> Response:
+        try:
+            start, end = get_date_range_from_params(request.GET)
+            interval = self._interval_from_request(request)
+            metrics_queries_plan = self._metrics_queries_plan_from_request(request)
+
+            results = run_metrics_queries_plan(
+                metrics_queries_plan=metrics_queries_plan,
+                start=start,
+                end=end,
+                interval=interval,
+                organization=organization,
+                # TODO: figure out how to make these methods work with HTTP body.
+                projects=self.get_projects(request, organization),
+                environments=self.get_environments(request, organization),
+                referrer=Referrer.API_DDM_METRICS_QUERY.value,
```

Comment:
Created a different referrer to separately track in snuba the usage of the new endpoint.